lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -560,6 +560,27 @@ export class UserAgentApplication {
         });
     }
 
+    /**
+     * API interfacing idToken request when applications already have a session/hint acquired by authorization client applications
+     * @param request
+     */
+    ssoSilent(request: AuthenticationParameters): Promise<AuthResponse> {
+        // throw an error on an empty request
+        if (!request) {
+            throw ClientConfigurationError.createEmptyRequestError();
+        }
+
+        // throw an error on no hints passed
+        if(!request.sid || !request.loginHint) {
```

Comment:
I thought there were valid use cases for this? e.g. guest accounts

Reply:
Reopening...what about apps that want to try to see if there is an existing session, but don't know about the users login hint? If we are going to tell them to call `acquireTokenSilent`, we might as well allow them to call `ssoSilent`, I would argue.

Reply:
We are not allowing them to call `acquireTokenSilent` today, we fail with "User login required". This is the same use case I pitched in the PR review and it seems to be shot down by the team. Even SPO use cases are not supported, they agreed to send `sid` and are in the process of rolling out those changes.

Reply:
Ah right, forgot about that.

Reply:
I would argue since there is already a check in `acquireTokenSilent`, you don't need this check, especially since its more complicated than just not having a `sid` or `loginHint`.

Reply:
The one benefit of having this is specific error is that you get disambiguation between silent and ats errors but I am fine either way

Reply:
@DarylThayil That's fair, but since the check in ATS is more complicated, maybe have that factored to a helper function, so the same logic can be used here?

Reply:
Extracting a helper will not be distinguishing between `ssoSilent` and `acquireTokenSilent` calls. Do we want to go that route?

Reply:
I modified the `acquireTokenSilent` error as: `User login is required. For silent calls, request must contain either sid of login_hint`. I still think that separating error messages for these two APIs even for the same root cause is a better user experience. Let us close on this asap.

Reply:
Okay, lets chat about this tomorrow afternoon.

Reply:
Currently, we allow not passing `login_hint` if there is already logged in user (which would silently fetch/renew the existing id token for the currently logged in user). I would argue we should still allow that, thoughts?